### PR TITLE
Issue #3446 reference version missing from parameter hash

### PR DESCRIPTION
**Pull request by punktilious**
Includes fixes for the following issues:
* #3446 fixes computation of parameter hash for fields using ByteBuffer
* #3416 prevent payload configuration falling back to default tenant
* #3439 handle null data after version-specific erase

### CODE REVIEW

File: fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/ParameterHashVisitor.java [hunk 1]
```diff
@@ -98,6 +98,7 @@ public void visit(TokenParmVal param) throws FHIRPersistenceException {
         updateDigestWithParmValKey(param);
         updateDigestWithString(param.getValueSystem());
         updateDigestWithString(param.getValueCode());
+        updateDigestWithString(param.getVersion());
```
[the remark this answers is not in the document]
Comment (punktilious) [reply fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/ParameterHashVisitor.java:101]: Yeah, should've done this. I'll add something. We probably need a rich set of tests around this anyway.

File: fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/ParameterHashTest.java [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import static org.testng.Assert.assertNotNull;
```
Comment (tbieste) [inline fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/ParameterHashTest.java:12]: Update copyright year.

Comment (punktilious) [reply fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/ParameterHashTest.java:12]: fixed

**Review by tbieste — APPROVED** [review]
LGTM

**Pull request merged**